### DISCUSSION TITLE
How do I add a hover button in a text editor？

### DISCUSSION THREAD

**Discussion by Anthonywck** [microsoft/vscode-discussions#2396] [Extension Development QnA] [answered]
**How to add a hover button at text editor when I select some text?** 

I need to develop a vscode extension. One of the features is that when I select a piece of code in the text editor, a button will hover, and clicking the button will trigger a command. How to achieve this?

I have found similar function buttons in cursor，how to achieve this  feature?
![image](https://github.com/user-attachments/assets/5286d3cc-6ab1-4546-aeab-6909c110fb70)

**Comment by alefragnani** [comment, marked as answer]
Hi @Anthonywck ,

There is no API to do exactly what you want to achieve, but here are a couple of options that I think worth a try.

I’m not sure it would work for _selected text_, but you may take a look a the `registerHoverProvider`, providing a `MarkdownString` in its result. It won’t give you a `Button` per se, but using links, attaching commands to it, may give you the same feeling.

Another alternative would be using Code Actions, as [this sample](https://github.com/microsoft/vscode-extension-samples/tree/main/code-actions-sample), but I guess, based on your scenario, it wouldn’t be the best approach.

Hope this helps
